### DISCUSSION TITLE
static method or riverpod provider?

### DISCUSSION THREAD

**Discussion by BigGitWorld** [rrousselGit/riverpod#1464] [General]
Hi.
In my projects a scenario occurs often, where I have to do something without keeping state and I don't need state management in this case.  so I can use one of the below solutions:

1. A normal class with static methods that receive ``ref`` as their argument and do their work.
2. A Provider variable with riverpod that keep a class.

solution(1) example:
```
class SampleStaicClassController {
  
  static void doSomething(Ref ref, other arguments){
    ref.read(a_riverpod_provider).(...);
  }
}

```

solution (2) example:
```
final sampleProvider = Provider((ref) => _SampleClassController(ref.read));
class _SampleClassController {

  final Reader _read;

  _SampleClassController(this._read);

  void doSomething(other arguments){
    _read(a_riverpod_provider).(...);
  }
}
```

Which one is better?

**Comment by rrousselGit** [comment]
Use 2)

Although instead of Reader, pass the ref itself

**Comment by BigGitWorld** [comment]
I agree.
